lib/server/services/qualityControlFlag/QCFlagTypesService.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { repositories: { QCFlagTypeRepository } } = require('../../../database');
+const { dataSource } = require('../../../database/DataSource.js');
+const { qcFlagTypeAdapter } = require('../../../database/adapters');
+const { BadParameterError } = require('../../errors/BadParameterError');
+const { NotFoundError } = require('../../errors/NotFoundError');
+
+/**
+ * @typedef QCFlagTypeIdentifier
+ * @param {number} id
+ * @param {string} name
+ */
+
+/**
+ * Quality control flags service
+ */
+class QCFlagTypesService {
+    /**
+     * Find an Quality Control Flag Type by its identifier
+     * @param {QCFlagTypeIdentifier} identifier identifier of Quality Control Flag Type
+     * @return {QCFlagType} a Quality Control Flag Type
+     */
+    async getByIdentifier({ id, name }) {
+        const queryBuilder = this.prepareQueryBuilder();
+
+        if (id !== null && id !== undefined) {
+            queryBuilder.where('id').is(id);
+        } else if (name) {
+            queryBuilder.where('name').is(name);
+        } else {
```

Comment:
Are you sure that you will have the case for name?
In any case, never write dead code because it may be used sometime.

Reply:
I put it the same way as for Period, data passes, simulation passes, as name is unique identifier, but for now it, in code, getting by name is not used, ok, to be cherry picked